src/components/FileTreeNode/index.spec.tsx

```diff
@@ -178,4 +247,177 @@ describe(__filename, () => {
+    (type, className, icon) => {
+      const message = {
+        ...fakeExternalLinterMessage,
+        file: 'manifest.json',
+        type: type as ExternalLinterMessage['type'],
+      };
+
+      const root = renderWithLinterMessage({
+        message,
+        treefoldRenderProps: {
+          isFolder: true,
+          getToggleProps: jest.fn(),
+        },
+      });
+
+      expect(root.find(className)).toHaveLength(1);
+
+      const nodeIcons = root.find(`.${styles.nodeIcons}`);
+      expect(nodeIcons).toHaveLength(1);
+      expect(nodeIcons.find(FontAwesomeIcon)).toHaveProp('icon', icon);
+      expect(nodeIcons.find(FontAwesomeIcon).prop('title')).toMatch(
+        new RegExp(`contains files with linter ${type}s`),
+      );
+    },
+  );
+
+  describe('findMostSevereTypeForPath', () => {
+    const _getMessageMap = (
+      messages: Partial<ExternalLinterMessage>[] = [fakeExternalLinterMessage],
+    ) => {
+      return getMessageMap(createFakeExternalLinterResult({ messages }));
+    };
+
+    it('returns null when the map is empty', () => {
+      const path = 'file.js';
+      const map = _getMessageMap([]);
+
+      const type = findMostSevereTypeForPath(map, path);
+
+      expect(type).toEqual(null);
+    });
+
+    it('returns the type of the message when there is only one message and a path', () => {
+      const path = 'file.js';
+      const message = { ...fakeExternalLinterMessage, file: path };
+      const map = _getMessageMap([message]);
+
+      const type = findMostSevereTypeForPath(map, path);
+
+      expect(type).toEqual(message.type);
+    });
+
+    it('returns null when the path is not found', () => {
+      const path = 'file.js';
+      const message = { ...fakeExternalLinterMessage, file: path };
+      const map = _getMessageMap([message]);
+
+      const type = findMostSevereTypeForPath(map, 'other-path');
+
+      expect(type).toEqual(null);
+    });
+
+    it('returns the most severe type given a directory path', () => {
+      const path = 'src';
+      const message = { ...fakeExternalLinterMessage, file: `${path}/file.js` };
+      const map = _getMessageMap([message]);
+
+      const type = findMostSevereTypeForPath(map, path);
+
+      expect(type).toEqual(message.type);
+    });
+
+    it('returns the most severe type given a map of messags and a directory path', () => {
+      const path = 'src';
+      const messages = [
+        {
+          ...fakeExternalLinterMessage,
+          file: `${path}/file-1.js`,
+          type: 'warning',
+        },
+        {
+          ...fakeExternalLinterMessage,
+          file: `${path}/file-2.js`,
+          type: 'error',
+        },
+      ] as ExternalLinterMessage[];
+      const map = _getMessageMap(messages);
+
+      const type = findMostSevereTypeForPath(map, path);
+
+      expect(type).toEqual(messages[1].type);
+    });
+
+    it('returns the most severe type given a map of messags and a full qualified file path', () => {
```

Comment:
As far as I can tell there isn't a test to cover multiple messages at the same exact path